clients/venice-producer/src/main/java/com/linkedin/venice/producer/online/ProducerTool.java

```diff
@@ -193,17 +196,23 @@ private static void writeToStore(ProducerContext producerContext) throws Excepti
 
   private static Object getValueObject(String valueString, RouterBasedStoreSchemaFetcher schemaFetcher) {
     Object value = null;
+    List<Exception> exceptionList = new ArrayList<>();
     for (Schema valueSchema: schemaFetcher.getAllValueSchemas()) {
```

Comment:
not path of this change, can we match the schemas in reverse order ? hope is that it will reduce older schema mismatches.

Reply:
`schemaFetcher.getAllValueSchemas()` returns a `Set` and not a `List`. So, we don't have info about the schema id there